src/commands/edge_app.rs

```diff
@@ -0,0 +1,241 @@
+        // We can let it crash in run time otherwise.
+        let updated_manifests: Vec<EdgeAppManifest> = serde_json::from_str(&response.text()?)?;
+
+        // overwrite manifest file now that we have id and extra fields.
+        let yaml = serde_yaml::to_string(&updated_manifests[0])?;
+        let input = File::create(path)?;
+        write!(&input, "{yaml}")?;
+
+        Ok(updated_manifests[0].clone())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::authentication::Config;
+
+    use envtestkit::lock::lock_test;
+    use envtestkit::set_env;
+    use httpmock::Method::POST;
+    use httpmock::MockServer;
+    use std::ffi::OsString;
+    use tempdir::TempDir;
+
+    #[test]
+    fn test_edge_app_init_should_create_screenly_yml() {
+        let tmp_dir = TempDir::new("test").unwrap();
+        let _lock = lock_test();
+        let _test = set_env(OsString::from("HOME"), tmp_dir.path().to_str().unwrap());
+        fs::write(tmp_dir.path().join(".screenly").to_str().unwrap(), "token").unwrap();
+
+        // init should make no requests so it's whatever for server url
+        let config = Config::new("asdf".to_owned());
+        let authentication = Authentication::new_with_config(config);
+        let command = EdgeAppCommand::new(authentication);
+
+        let p = tmp_dir.path().join("screenly.yml");
+        assert!(command.init(Path::new(p.to_str().unwrap())).is_ok());
+
+        assert_eq!(
+            EdgeAppManifest::default(),
+            serde_yaml::from_str(&fs::read_to_string(Path::new(p.to_str().unwrap())).unwrap())
+                .unwrap()
+        );
+    }
+
+    #[test]
+    fn test_edge_app_publish_should_send_correct_request() {
+        let tmp_dir = TempDir::new("test").unwrap();
+        let _lock = lock_test();
+        let _test = set_env(OsString::from("HOME"), tmp_dir.path().to_str().unwrap());
+        fs::write(tmp_dir.path().join(".screenly").to_str().unwrap(), "token").unwrap();
+        let manifest = EdgeAppManifest {
+            id: "".to_string(),
+            created_at: "".to_string(),
+            created_by: "".to_string(),
+            permissions: vec![],
+            updated_at: "".to_string(),
+
+            name: "Test".to_string(),
+            version: "100".to_string(),
+            description: "Best".to_string(),
+            icon: "?".to_string(),
+            author: "Best author ever".to_string(),
+            homepage_url: "test.io".to_string(),
+        };
+
+        let published_manifest = vec![EdgeAppManifest {
+            id: "01GS5H2CX6Y10ZRJHEDQPEWN4E".to_string(),
+            created_at: "2023-02-13T13:57:43+00:00".to_string(),
+            created_by: "01D0C6YJA8000A7JEN0230MWGF".to_string(),
+            updated_at: "2023-02-13T13:57:43.97356+00:00".to_string(),
+            permissions: vec![],
+
+            ..manifest.clone()
+        }];
+
+        let mut b = serde_json::to_value(&manifest).unwrap();
```

Comment:
Use normal variable names please.

Reply:
why? it's test code. Letter b should be fine for variables whose whole purpose is to hold a temporary value (binding).